cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx

```diff
@@ -188,26 +220,44 @@ function VmTemplateSection(
     vmTemplates: VSphereVirtualMachine[],
     errors: { [key: string]: FieldError | undefined },
     register: any,
+    osImagesLoading: boolean,
+    handleRefresh: (event: MouseEvent<HTMLAnchorElement>) => void,
     onOsImageSelected: (osImage: string, field: VSPHERE_CLUSTER_SETTING_STEP_FIELDS) => void
 ) {
     const handleOsImageSelect = (event: ChangeEvent<HTMLSelectElement>) => {
         onOsImageSelected(event.target.value || '', field);
     };
+
     const fieldError = errors[field];
     return (
-        <div cds-layout="m:lg">
-            <CdsSelect layout="vertical" controlWidth="shrink">
-                <label>OS Image</label>
-                <select {...register(VSPHERE_FIELDS.VMTEMPLATE)} onChange={handleOsImageSelect}>
+        <CdsFormGroup layout="vertical-inline" control-width="shrink">
+            <div cds-layout="horizontal gap:lg align:vertical-center">
+                <SpinnerSelect
+                    className="select-md-width"
+                    disabled={false}
+                    label="OS Image"
+                    handleSelect={handleOsImageSelect}
+                    name={VSPHERE_FIELDS.VMTEMPLATE}
+                    controlMessage="OS Images will be retrieved after selecting Datacenters"
```

Comment:
nitpick: please change to "OS Images will be retrieved after selecting a Datacenter"

Reply:
Will do